### DISCUSSION TITLE
Change Object Type

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [fabricjs/fabric.js#7737] [Q&A] [answered]
Hey there, :)

I know this question might be a bit silly, obvious, and unnecessary, but is there any way to change the "type" of the object programmatically?

For example, imagining that I created a rectangle, but when the user clicks on a button the object is no longer a rectangle and changes to a circle.

I have a [codepen](https://codepen.io/201flaviosilva/pen/WNXKRbm) with code I want to make: https://codepen.io/201flaviosilva/pen/WNXKRbm

I think I could create 2 objects (a circle and a rectangle) and by clicking on the button, toggle the visibility, but the problem is if the application starts to grow it can start to be a performance problem, and I would like to have support for more objects. :)

(Sorry, for my English :))

**Comment by asturur** [comment, marked as answer]
Well there isn't a way to change object type, many object have incompatible properties.
What you can do is do that on request instead of changing visibility.
When the button is clicked, you get the active object, you create a new one and you swap them, removing the other.


```js
const selected = canvas.getActiveObject();
const newObject = new fabric.Circle({ ... add properties reading from selected as your app need });
canvas.remove(selected);
canvas.add(newObject);
```

There isn't much else to do.

**Reply by 201flaviosilva** [reply]
Yes, basically I ended up doing it that way.
Thank you very much for the answer, any way :)
